### PR TITLE
Task/APPS-2570 – staff members for assigning tasks

### PR DESCRIPTION
**Pull request by giuliacampana**
- when app is initialized, get all staff entities
- then, get all users who have READ role in the app
- then, create a new entity for any users who do not have a corresponding entity in staff entity set

### CODE REVIEW

File: src/containers/app/AppSagas.js [hunk 1]
```diff
@@ -2,34 +2,159 @@
...
+    let response = yield call(getEntitySetDataWorker, getEntitySetData({ entitySetId: staffESID }));
+    if (response.error) throw response.error;
+    let staff :List = fromJS(response.data);
+
+    const orgId = selectedAppConfig.organization?.id;
+    response = yield call(getOrganizationWorker, getOrganization(selectedAppConfig.organization?.id));
+    if (response.error) throw response.error;
+
+    const readRole :?Object = response.data?.roles?.find((role) => role.title?.includes('READ'));
+    const readRoleId :?UUID = readRole?.id;
```
Comment (anzioka) [inline src/containers/app/AppSagas.js:90]: please consider using PermissionTypes constants from lattice: https://github.com/openlattice/lattice-js/blob/develop/src/constants/types/PermissionTypes.js

```
import { Types } from 'lattice

const { PermissionTypes } = Types

PermissionTypes.READ
```

File: src/containers/app/AppSagas.js [hunk 2]
```diff
@@ -2,34 +2,159 @@
...
+    const authorizedUserEmails = authorizedUsers.map((user) => user.email);
+
+    const entityData = {
+      [staffESID]: []
+    };
+
+    staff.forEach((staffMember) => {
+      const userEmail = getPropertyValue(staffMember, [COUNTY_ID, 0]);
+      if (!authorizedUserEmails.includes(userEmail)) {
+        const user = authorizedUsers.find((authorizedUser) => authorizedUser.email === userEmail);
```
Comment (anzioka) [inline src/containers/app/AppSagas.js:109]: This `!authorizedUserEmails.includes(userEmail)` looks like a duplicate. You only need to the next lines Set `find` call and check for undefined

File: src/containers/app/AppSagas.js [hunk 3]
```diff
@@ -2,34 +2,159 @@
...
+        const userFirstName = user.given_name;
+        const userLastName = user.family_name;
+        const { email } = user;
+        entityData[staffESID].push({
+          [personGivenNamePTID]: [userFirstName],
+          [personSurnamePTID]: [userLastName],
+          [idPTID]: email,
```
Comment (anzioka) [inline src/containers/app/AppSagas.js:116]: just curious why property multiplicity on the firstname & lastname but not email.

File: src/containers/app/AppSagas.js [hunk 4]
```diff
@@ -2,34 +2,159 @@
...
+    yield put(getCurrentStaff.request(action.id));
+    const { selectedAppConfig, staffESID } = action.value;
+
+    let response = yield call(getEntitySetDataWorker, getEntitySetData({ entitySetId: staffESID }));
+    if (response.error) throw response.error;
+    let staff :List = fromJS(response.data);
+
+    const orgId = selectedAppConfig.organization?.id;
+    response = yield call(getOrganizationWorker, getOrganization(selectedAppConfig.organization?.id));
+    if (response.error) throw response.error;
```
[the remark this answers is not in the document]
Comment (giuliacampana) [reply src/containers/app/AppSagas.js:87]: oops thanks!

File: src/containers/app/AppSagas.js (hunk 1 above)
Comment (giuliacampana) [reply src/containers/app/AppSagas.js:90]: ah thank you—will do

File: src/containers/app/AppSagas.js (hunk 3 above)
Comment (giuliacampana) [reply src/containers/app/AppSagas.js:116]: oops this is a mistake, thanks for catching it

File: src/containers/app/AppSagas.js (hunk 2 above)
Comment (giuliacampana) [reply src/containers/app/AppSagas.js:109]: oh good point. this logic makes more sense, thanks!

**Review by anzioka — APPROVED** [review]

**Review by bergieboy — APPROVED** [review]

**Pull request merged**